packages/event/event.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * @typedef {Object<string, Set<Function>>} Events
+ */
+
+/**
+ * @typedef SubscribeReturn
+ * @property {Function} unsubscribe - The unsubscribe function to unsubscribe the callback from the event
+ */
+
+/**
+ * The event module manages a list of subscribers and is responsible to send an event to the subscribers using pub/sub pattern
+ *
+ * @example
+ * // publish an event
+ * event.publish('stream:ice-connection-state-change', {
+     type: 'stream:ice-connection-state-change',
+     detail: {
+       iceConnectionState: 'connected',
+     },
+   })
+ *
+ * // subscribe to an event
+ * event.subscribe(
+      'stream:ice-connection-state-change',
+      (data) => {
+        // handle the event
+      }
+   )
+ */
+const event = (() => {
+  /**
+   * ======================================================
+   *  Variables
+   * ======================================================
+   */
+
+  /** @type {Events} */
+  const events = {}
+
+  /**
+   * ======================================================
+   *  Functions
+   * ======================================================
+   */
+
+  /**
+   *
+   * @param {Events} events - The event object that holds all events and callbacks for each event
+   * @param {string} eventName - The name of the event
+   * @returns {boolean} - Returns true if the events[eventName] value is instanceOf Set
+   */
+  const isInstanceOfSet = (events, eventName) =>
+    events[eventName] instanceof Set
+
+  /**
+   * Publish an event and trigger the callback
+   *
+   * @param {string} eventName - The name of the event
+   * @param {any} data - The actual data sent
+   */
+  const publish = (eventName, data) => {
+    if (isInstanceOfSet(events, eventName)) {
+      for (const callback of events[eventName]) {
+        callback(data)
+      }
+    } else {
+      events[eventName] = new Set()
+    }
+  }
+
+  /**
+   * Enable callback function to subscribe to specific event
+   *
+   * @param {string} eventName - The name of the event
+   * @param {Function} callback - A callback function
+   * @returns {SubscribeReturn} An object that contains the unsubscribe function
+   */
+  const subscribe = (eventName, callback) => {
+    if (!isInstanceOfSet(events, eventName)) {
+      events[eventName] = new Set()
+    }
+    events[eventName].add(callback)
+
+    /**
+     * Enable callback function to unsubscribe from specific event
+     */
+    const unsubscribe = () => {
+      events[eventName].delete(callback)
+    }
+
+    return {
+      unsubscribe,
+    }
```

Comment:
@faiq-naufal sorry, I'm just confused on this part. Why do we need the unsubscribe part here? Could you help to explain it? ^^"

Reply:
To subscribe to an event, the subscribers need to provide a callback function that can be run by the SDK when an event is published. They can get the data from the published event and process the data through the callback function.  

When a subscriber already subscribes to the event, every time the event is published, the SDK will run the callback function the subscriber already provided again and again. What if the subscriber only wants to call the callback function once or twice? They don't want to subscribe to the event continuously because maybe they don't need anymore to subscribe to the event in their logic. This is when the `unsubscribe` function appear. Basically this unsubscribe will delete the callback function that is already subscribed to the SDK so the SDK will not run the callback function again when the event is published

I forgot to put the unsubscribe example in the description. To unsubscribe the event, you can do this
```js
// subscribe to an event
  const subscriber = event.subscribe(
    'stream:ice-connection-state-change',
    (data) => {
      // handle the event
    }
  )
  
  subscriber.unsubscribe()
```

Reply:
@faiq-naufal oh, okay it makes sense. Thank you for the explanation mas!